conda/models/match_spec.py

```diff
@@ -177,8 +178,8 @@ def __init__(self, optional=False, target=None, **kwargs):
     @classmethod
     def from_dist_str(cls, dist_str):
         parts = {}
-        if dist_str.endswith(CONDA_TARBALL_EXTENSION):
-            dist_str = dist_str[:-len(CONDA_TARBALL_EXTENSION)]
+        if dist_str.endswith(CONDA_PACKAGE_EXTENSION_V1):
+            dist_str = dist_str[:-len(CONDA_PACKAGE_EXTENSION_V1)]
```

Comment:
Should only be used in deprecated code paths.  `CONDA_PACKAGE_EXTENSION_V1` should be fine here.